lib/test/karma.js

```diff
@@ -16,7 +18,69 @@ function quote(value) {
   return '"' + value + '"';
 }
 
+var defaultReporterName = 'angularity-karma-reporter';
+
 var filesAppendRegex = /\/\*+\s*ANGULARITY_FILE_LIST\s*\*+\// ;
+var reportersAppendRegex = /\/\*+\s*ANGULARITY_REPORTER_LIST\s*\*+\// ;
+var pluginsAppendRegex = /\/\*+\s*ANGULARITY_PLUGIN_LIST\s*\*+\// ;
+var karmaReporterMatchNameRegex = /^karma-(.+)-reporter$/ ;
+
+/**
+ * Determine the `require` path for a plugin.
+ *
+ * @param {string} reporterName Either a package name,
+ *   *OR* an absolute path, for a karma reporter.
+ * @return {string} In karma config file, `plugins` will be inline `require`d
+ *   using this return value.
+ */
+function getKarmaReporterPluginPath(reporterName) {
+  if (typeof reporterName !== 'string') {
+    throw 'Get Karma Reporter Plugin Path: Reporter name is unspecified';
+  }
+
+  if (reporterName === defaultReporterName) {
+    return defaultReporterName;
+  }
+  else {
+    var reporterPath = (path.dirname(reporterName) === '.') ?
+      path.resolve('node_modules', 'karma-'+reporterName+'-reporter') :
+      reporterName;
+    try {
+      require(reporterPath);
+    }
+    catch (ex) {
+      throw 'Get Karma Reporter Plugin Path: Attempt to require reporter from path '+reporterPath+' with no success.';
+    }
+    return reporterPath;
+  }
+}
+
+/**
+ * Determine the registered reporter name.
+ *
+ * @param {string} reporterName Either the registered reporter name,
+ *   *OR* an absolute path, for a karma reporter.
+ * @return {string} The registered reporter name for use in
+ *   the `reporters` section of the karma config file.
+ */
+function getKarmaReporterName(reporterName) {
+  if (typeof reporterName !== 'string') {
+    throw 'Get Karma Reporter Name: Reporter name is unspecified';
+  }
+
+  var name;
+  if (path.dirname(reporterName) === '.') {
+    name = reporterName;
+  }
+  else {
+    name = path.basename(reporterName);
+  }
+  var match = karmaReporterMatchNameRegex.exec(name);
+  if (!!match && match.length === 2) {
```

Comment:
- Aids readability - another developer loking at the code will know it is supposed to be a Boolean value
- Aids optimisation techniques - the JS interpreter can more easily optimise this because it knows it is supposed to be a Boolean value too